### DISCUSSION TITLE
Add high scores option to SuperTux

### DISCUSSION THREAD

**Discussion by tobbi** [SuperTux/supertux#3317] [Ideas]
From Twitter: https://twitter.com/petittux/status/740671016356175872

I wonder whether we should add a high scores option to SuperTux. We could then allow people to enter their name etc. Might be fun if you're playing SuperTux with more than one player.

**Comment by brmbrmcar** [comment]
I think the profiles should also be done by custom name, with the ability to add more.

**Comment by tobbi** [comment]
This is not what this issue is about.

**Reply by borissecret** [reply]
Не знам как става. ако някой преведе това на английски, може ли да ми помогне?
И без това съм напреднал в програмирането и в играта, но C++ не знам, знам само C.

**Reply by provigz** [reply]
@borissecret С какво точно ти е нужна помощ? Между другото, тази функция за най-добри резултати не е все още планирана, а само е сложена за дискусия тук.

Също, ако знаеш C, би трябвало лесно да се приспособиш и към C++.

**Reply by borissecret** [reply]
```
Здравейте, благодаря че ми отговорихте. 
Да така е, но някой тънкости на C++ не ги знам, даже изобщо не го знам, но все пак C ми е интересен, но само ако програмирам без графики, защото тях не ги харесвам и не ги умея.
Аз сега съм във втори клас и знам много неща и ми е интересно програмирането и Tux. Не ми харесва на scratch, защото има картинки и компютъра на мама загрява.
Ако имате време, можете да ми кажете поне какво да прочета на български.
Ето пример какво мога да напиша на С, мъничко е:

#include <stdio.h>
#include <stdlib.h>
#define r = rand()

int main() {
    while(r < 100) {
        printf(r);
        
        r = null;
        r = rand();
    }
    if (r > 100) {
        printf("Програмата завърши!");
    }
    return 0;
}

```

**Reply by provigz** [reply]
@borissecret Страхотно е, че изпробваш работа с езици като C от ранна възраст - един ден ще ти бъде доста полезно това! Въпреки че даденият код не се компилира и има основни грешки, това че имаш интерес е супер! Обаче, за работа с цели проекти като SuperTux се изискват фундаментални знания на C++ и като цяло по програмиране. Все още ти е най-вероятно рано. В бъдеще, дори и да не си толкова спец по програмиране може да се упражняваш като работиш по някои лесни проблеми или функции за SuperTux (аз така започнах със C++ например).

Сигурно ще учите скрач в 3-ти клас? Дори и да не ти харесва в момента, дай му отново шанс - страшно е полезен като за начало в програмирането.

**Comment by brmbrmcar** [comment]
Right! I have had it!

**Comment by Karkus476** [comment]
@brmbrmcar Feel free to make a new issue, or comment on one which already addresses your issue. It's a good idea. 
@tobbi Do you mean high scores of coins, time & badguys etc. ?

**Comment by tobbi** [comment]
@Karkus476 You're right, maybe I should've thought about this some more. We need a way to compare high scores against each other. Or maybe only use time as an indicator.

**Comment by Karkus476** [comment]
@tobbi Time as an indicator is okay, but not as impressive as assigning a name to your high score

**Comment by tobbi** [comment]
Yes, but how do we _rank_ highscores in the first place...That's what I meant.

**Comment by Karkus476** [comment]
@tobbi I see... Maybe different 'tabs' could show the 'top' ten for each variable: lowest time, highest badguys, highest coins

**Comment by tobbi** [comment]
That could work, however, a lot of people will have the "highest badguys" and "highest coins", because those are finite, and there can only be one ideal score.

**Comment by Alasdairbugs** [comment]
Moved to discussions to get more ideas, this bumps up the idea to the top upon more people contributing ideas.

**Comment by borissecret** [comment]
Еми, не съм го написал правилно, като запомня как ставаше, ще ти го пратя
наново. Кажи ми обаче какви грешки дава компилатора. Иначе се опитвам да
направя тази работа.

**Comment by borissecret** [comment]
Добре също, разбирам и защо тука е писан с С++, защото е обектно-ориентиран
и се използва по лесно за графиките.

**Comment by borissecret** [comment]
Може github да допуска грешки, ще си спра авторедактора.

**Reply by provigz** [reply]
Тази дискусия е извън темата тук и трябва да спре вече. Иначе ползвай онлайн C компилатор за да видиш грешките.

This discussion is off-topic and should now end. By the way, you can use an online C compiler to check the errors in your code.

**Comment by bruhmoent** [comment]
Моля, общувайте само на английски език.
